### PR TITLE
introduce WellInterfaceGeneric::computeWellPotentials and checkNegativeWellPotentials

### PR DESCRIPTION
**Pull request by akva2**
now code is shared between StandardWell and MultisegmentWell

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/wells/WellInterfaceGeneric.hpp [hunk 1]
```diff
@@ -214,6 +214,14 @@ class WellInterfaceGeneric {
     bool wellUnderZeroRateTarget(const SummaryState& summary_state,
                                  const WellState& well_state) const;
 
+    std::tuple<bool,bool>
```
Comment (bska) [inline opm/simulators/wells/WellInterfaceGeneric.hpp:217]: Just out of curiosity: Why `tuple<>` instead of `pair<>` here?

Comment (akva2) [reply opm/simulators/wells/WellInterfaceGeneric.hpp:217]: no reason, changed.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the update.  I'll merge into master.

**Pull request merged**